scripts/bytecodecompare/prepare_report.py

```diff
@@ -51,10 +64,19 @@ def load_source(path: Union[Path, str]) -> str:
 def parse_standard_json_output(source_file_name: Path, standard_json_output: str) -> FileReport:
     decoded_json_output = json.loads(standard_json_output.strip())
 
+    # JSON interface still returns contract metadata in case of an internal compiler error while
+    # CLI interface does not. To make reports comparable we must force this case to be detected as
+    # an error in both cases.
```

Comment:
Why is this done here and in the JS code?

Reply:
It is in both. See `if (['UnimplementedFeatureError', 'CompilerError', 'CodeGenerationError'].includes(error['type']))` line in `prepare_report.js`.

Reply:
That's what I mean, why is it done in both? Do they run in different places?

Reply:
Sorry, I misread your question. Are you asking why if's done in JS version as well? There are four combinations:
1. `solc --standard-json`
2. `solc`
3. emscripten

The JS version is 3. while the Python scripts handles 1. and 2. This `if` is needed for 1. and 3. to make it fail just like 2.

Reply:
Ok!